compute/instances/spot/create_spot.go

```diff
@@ -0,0 +1,140 @@
+	if err != nil {
+		return nil, fmt.Errorf("NewImagesRESTClient: %w", err)
+	}
+	defer client.Close()
+
+	req := &computepb.GetFromFamilyImageRequest{
+		Project: project,
+		Family:  family,
+	}
+
+	resp, err := client.GetFromFamily(ctx, req)
+	if err != nil {
+		return nil, fmt.Errorf("GetFromFamily: %w", err)
+	}
+	return resp, nil
+}
+
+// diskFromImage creates an AttachedDisk object to be used in VM instance creation using an image as the source.
+func diskFromImage(diskType string, diskSizeGb int64, boot bool, sourceImage string, autoDelete bool) *computepb.AttachedDisk {
+	return &computepb.AttachedDisk{
+		AutoDelete: proto.Bool(autoDelete),
+		Boot:       proto.Bool(boot),
+		InitializeParams: &computepb.AttachedDiskInitializeParams{
+			DiskSizeGb:  proto.Int64(diskSizeGb),
+			DiskType:    proto.String(diskType),
+			SourceImage: proto.String(sourceImage),
+		},
+		Type: proto.String(computepb.AttachedDisk_PERSISTENT.String()),
+	}
+}
+
+// createInstance sends an instance creation request to the Compute Engine API and waits for it to complete.
+func createInstance(ctx context.Context, projectID, zone, instanceName string, disks []*computepb.AttachedDisk, machineType, networkLink string, spot bool) (*computepb.Instance, error) {
+	client, err := compute.NewInstancesRESTClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("NewInstancesRESTClient: %w", err)
+	}
+	defer client.Close()
+
+	networkInterface := &computepb.NetworkInterface{
+		Name: proto.String(networkLink),
+	}
+
+	instance := &computepb.Instance{
+		Name:              proto.String(instanceName),
+		Disks:             disks,
+		MachineType:       proto.String(fmt.Sprintf("zones/%s/machineTypes/%s", zone, machineType)),
+		NetworkInterfaces: []*computepb.NetworkInterface{networkInterface},
+		Scheduling: &computepb.Scheduling{
+			ProvisioningModel: proto.String(computepb.Scheduling_SPOT.String()),
+		},
+	}
+
+	req := &computepb.InsertInstanceRequest{
+		Project:          projectID,
+		Zone:             zone,
+		InstanceResource: instance,
+	}
+
+	op, err := client.Insert(ctx, req)
+	if err != nil {
+		return nil, fmt.Errorf("Insert: %w", err)
+	}
+
+	opClient, err := compute.NewZoneOperationsRESTClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("NewZoneOperationsRESTClient: %w", err)
+	}
+	defer opClient.Close()
+
+	if err = op.Wait(ctx); err != nil {
+		return nil, fmt.Errorf("unable to wait for the operation: %w", err)
+	}
+
+	return client.Get(ctx, &computepb.GetInstanceRequest{
+		Project:  projectID,
+		Zone:     zone,
+		Instance: instanceName,
+	})
+}
+
+// createSpotInstance creates a new Spot VM instance with Debian 10 operating system.
+func createSpotInstance(w io.Writer, projectID, zone, instanceName string) error {
+	// projectID := "your_project_id"
+	// zone := "europe-central2-b"
+	// instanceName := "your_instance_name"
+
+	ctx := context.Background()
+	image, err := getImageFromFamily("debian-cloud", "debian-11")
```

Comment:
these helper functions obscure the flow of code. please refactor this to inline your helper functions, so the code is more readable.

consider this related sample: https://github.com/GoogleCloudPlatform/golang-samples/blob/main/compute/instances/create-start-instance/create_instance_from_public_image.go#L29